File: lectures/13/Books/app/src/main/java/ro/ubbcluj/cs/books/db/BookDatabase.java
package ro.ubbcluj.cs.books.db;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.TypeConverters;

import ro.ubbcluj.cs.books.domain.Book;

@Database(entities = {Book.class}, version = 1)
@TypeConverters({Converters.class})
public abstract class BookDatabase extends RoomDatabase {
  public abstract BooksDao getBookDao();
}

File: lectures/13/Books/app/src/main/java/ro/ubbcluj/cs/books/db/BookDatabase.kt
package ro.ubbcluj.cs.books.db

import android.arch.persistence.room.Database
import android.arch.persistence.room.RoomDatabase
import android.arch.persistence.room.TypeConverters

import ro.ubbcluj.cs.books.domain.Book

@Database(entities = [Book::class], version = 1)
@TypeConverters(Converters::class)
abstract class BookDatabase : RoomDatabase() {
  abstract val bookDao: BooksDao
}
